src/backend/rewrite/rewriteHandler.c

```diff
@@ -1411,6 +1413,19 @@ rewriteTargetListUD(Query *parsetree, RangeTblEntry *target_rte,
 						   type_mod,
 						   type_coll,
 						   0);
+
+		if (rel_is_partitioned(reloid))
```

Comment:
The problem is that service columns `ctid` and `gp_segment_id` are added to targetList at rewriting stage, and we need somehow to do that only when relation is partitioned. And yep, this code leads to `tableoid` be added for legacy planner as well.  I think that is an unneseccary move, because there is no need for that column for postgres's plans. 
```
explain (costs off, verbose)  delete from test where j = 3;
LOG:  statement: explain (costs off, verbose)  delete from test where j = 3;
                                            QUERY PLAN                                            
--------------------------------------------------------------------------------------------------
 Delete on public.test_1_prt_extra
   ->  Seq Scan on public.test_1_prt_extra
         Output: test_1_prt_extra.ctid, test_1_prt_extra.gp_segment_id, test_1_prt_extra.tableoid
         Filter: (test_1_prt_extra.j = 3)
   ->  Seq Scan on public.test_1_prt_3
         Output: test_1_prt_3.ctid, test_1_prt_3.gp_segment_id, test_1_prt_3.tableoid
         Filter: (test_1_prt_3.j = 3)
 Optimizer: Postgres query optimizer
 Settings: optimizer=off
(9 rows)
```
Probably we could avoid that by extending the targetList only when ORCA is hooked.